spec/requests/v1/padlock/registration_request_spec.rb

```diff
@@ -0,0 +1,120 @@
+require "rails_helper"
+
+RSpec.describe V1::Padlock::RegistrationController, type: :request do
+  describe 'registration' do
+    let(:request) { post registration_index_path, params: params }
+
+    let(:user_params) do
+      {
+        email:                 'example@gmail.com',
+        password:              '12345678',
+        password_confirmation: '12345678'
+      }
+    end
+
+    context 'success' do
+      let(:params) { { user: user_params } }
+      it "returns status 'created'" do
+        request
+
+        expect(response).to have_http_status :created
+      end
+
+      it 'creates a new user' do
+        expect{ request }.to change(User, :count).by(1)
+      end
+
+      it 'returns created user' do
+        request
```

Comment:
group specs with `before { request }` within context

Reply:
I'm using it inside the block to check if it  creates a user

Reply:
@andrewr224 
You can do the same within context with before :)